### PR TITLE
SCUMM: Bugfixes

### PR DESCRIPTION
**Pull request by BenCastricum**
Here's a proposal for fixing 3 bugs in SCUMM HE games. Feel free to cherry-pick. Feedback appreciated.

All fixes have been play-tested by me.

### CODE REVIEW

File: engines/scumm/scumm.cpp [hunk 1]
```diff
@@ -2589,6 +2589,31 @@ void ScummEngine_v60he::setHETimer(int timer) {
...
+void ScummEngine_v60he::pauseHETimers(bool pause) {
+	// The HE timers rely on system time which of course doesn't pause when
+	// the engine does. By adding the elapsed time we compensate for this.
+	// Fixes bug #6352
+	if (pause) {
+		// Pauses can be layered, we only need the start of the first
+		if (!_pauseStartTime)
+			_pauseStartTime = _system->getMillis();
+	}
+	else {
```
Comment (bluegr) [inline engines/scumm/scumm.cpp]: Code formatting

**Comment by bluegr** [comment]
Other than the few minor things noted, this looks great. Thanks for your work! :)

**Comment by BenCastricum** [comment]
@bluegr Thanks for reviewing. I fixed both issues.

**Comment by sev-** [comment]
Great work!

**Pull request merged**